Sources/Basic/SortedArray.swift

```diff
@@ -59,14 +68,64 @@ public struct SortedArray<Element>: CustomStringConvertible {
     
     /// Insert the given sequence, maintaining the sort order.
     public mutating func insert<S: Sequence>(contentsOf newElements: S) where S.Iterator.Element == Element {
+        var newElements = newElements.sorted(by: areInIncreasingOrder)
+        guard !newElements.isEmpty else {
+            return
+        }
+        guard !elements.isEmpty else {
+            elements = newElements
+            return
+        }
+
+        var lhsIndex = elements.endIndex - 1
+        var rhsIndex = newElements.endIndex - 1
+
+        elements.reserveCapacity(elements.count + newElements.count)
+
+        // NOTE: If SortedArray moves to stdlib an _ArrayBuffer can be used
+        // instead. This append can then be removed as _ArrayBuffer can be
+        // resized without requiring instantiated elements.
         elements.append(contentsOf: newElements)
-        elements.sort(by: areInIncreasingOrder)
+
+        // NOTE: Experimentally this seemed to be the inflection point where
+        // the constant overheads of this implementation outweighed the better
+        // O(n) versus O(n log(n)) complexity.
+        guard elements.count > 1000 else {
```

Comment:
About 2x faster, although in absolute terms perhaps not that much...? I will add a test for it.